client/js/widgets/widget.js

```diff
@@ -1025,12 +1025,41 @@ export class Widget extends StateManaged {
           for(const key in a.in)
             await callWithAdditionalValues({ key, value: a.in[key] }, {});
           if(jeRoutineLogging)
-            jeLoggingRoutineOperationSummary( `element in '${JSON.stringify(a.in)}'`);
+            jeLoggingRoutineOperationSummary( `elements in '${JSON.stringify(a.in)}'`);
+        } else if(a.range) {
+          let range = asArray(a.range);
+          if(range.length == 1)
+            range.unshift(1);
+          if(range.length == 2)
+            range.push(1);
```

Comment:
Perhaps push -1 if end < start?

Reply:
Yeah, no reason to set a default value that later generates a warning that the value got changed. I was also thinking there is some room for improvement in the step validation.

Reply:
Have another look.